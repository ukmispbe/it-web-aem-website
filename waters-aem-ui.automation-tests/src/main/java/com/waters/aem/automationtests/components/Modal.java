package com.waters.aem.automationtests.components;

import com.cognifide.qa.bb.qualifier.PageObjectInterface;


@PageObjectInterface
public interface Modal {

    String getText();

    String getIcon();

}
