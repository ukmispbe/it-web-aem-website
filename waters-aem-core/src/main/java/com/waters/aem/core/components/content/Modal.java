package com.waters.aem.core.components.content;

import com.adobe.cq.export.json.ComponentExporter;
import com.adobe.cq.export.json.ExporterConstants;
import com.citytechinc.cq.component.annotations.Component;
import com.citytechinc.cq.component.annotations.DialogField;
import com.citytechinc.cq.component.annotations.Tab;
import com.citytechinc.cq.component.annotations.widgets.PathField;
import com.citytechinc.cq.component.annotations.widgets.TextField;
import com.icfolson.aem.library.core.components.AbstractComponent;
import com.waters.aem.core.constants.WatersConstants;
import com.waters.aem.core.utils.LinkUtils;
import org.apache.sling.api.SlingHttpServletRequest;
import org.apache.sling.models.annotations.DefaultInjectionStrategy;
import org.apache.sling.models.annotations.Exporter;
import org.apache.sling.models.annotations.Model;

import javax.annotation.Nonnull;
import javax.inject.Inject;

@Component(value = "Modal",
    description = "This is the Modal component for Waters site",
    tabs = @Tab(title = "Properties"))
@Model(adaptables = SlingHttpServletRequest.class,
    adapters = { Modal.class, ComponentExporter.class },
    resourceType = Modal.RESOURCE_TYPE,
    defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME,
    extensions = ExporterConstants.SLING_MODEL_EXTENSION)
public class Modal extends AbstractComponent implements ComponentExporter {

    public static final String RESOURCE_TYPE = "waters/components/content/modal";

    @DialogField(fieldLabel = "Source Text",
        fieldDescription = "Enter the source text",
        required = true,
        ranking = 2)
    @TextField
    @Inject
    private String sourceText;

    @DialogField(fieldLabel = "Source Icon",
        fieldDescription = "Select source Icon",
        ranking = 2)
    @PathField(rootPath = WatersConstants.DAM_ICON_PATH)
    @Inject
    private String sourceIcon;

    @DialogField(fieldLabel = "Modal Heading Text",
        fieldDescription = "Enter the text modal heading",
        ranking = 3)
    @TextField
    @Inject
    private String modalHeadingText;

    @DialogField(fieldLabel = "Modal Heading Icon",
        fieldDescription = "Select modal heading Icon",
        ranking = 4)
    @PathField(rootPath = WatersConstants.DAM_ICON_PATH)
    @Inject
    private String modalHeadingIcon;

    @DialogField(fieldLabel = "Modal Body Text",
        fieldDescription = "Enter the modal body text",
        ranking = 5)
    @TextField
    @Inject
    private String modalBodyText;

    public String getSourceText() {
        return sourceText;
    }

    public String getSourceIcon() {
        return sourceIcon;
    }

    public String getModalHeadingText() {
        return modalHeadingText;
    }

    public String getModalHeadingIcon() {
        return modalHeadingIcon;
    }

    public String getModalBodyText() {
        return modalBodyText;
    }

    public String getUniqueId() {
        return getId();
    }

    public Boolean isSourceIconSvg() {
        return LinkUtils.isSvg(sourceIcon);
    }

    public Boolean isHeadingIconSvg() {
        return LinkUtils.isSvg(modalHeadingIcon);
    }

    @Nonnull
    @Override
    public String getExportedType() {
        return RESOURCE_TYPE;
    }
}
