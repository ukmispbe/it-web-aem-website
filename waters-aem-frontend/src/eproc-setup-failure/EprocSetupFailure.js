import React from 'react';
import PropTypes from 'prop-types';

import Modal, { Header, keys } from '../utils/modal';
import AddToCartBody from '../sku-details/views/addToCartModal';
import './styles/index.scss';

function EprocSetupFailure(props) {
    const { status, title, text, icon, buttons } = props;
    return (
        <Modal isOpen={status} onClose={() => { }} className='cmp-add-to-cart-modal' showCloseIcon={false}>
            <Header
                title={title}
                icon={icon}
                className={keys.HeaderWithAddedMarginTopError}
            />
            <AddToCartBody
                config={{
                    isOrderDetails: true,
                    textHeading: '',
                    text,
                    buttons
                }}
                errorObjCart={{}}
                onClose={() => { }}
            ></AddToCartBody>
        </Modal>
    );
}

EprocSetupFailure.propTypes = {
    status: PropTypes.bool,
    title: PropTypes.string,
    text: PropTypes.string,
    icon: PropTypes.string,
    buttons: PropTypes.array
}

EprocSetupFailure.defaultProps = {
    status: false,
    title: '',
    text: '',
    icon: '',
    buttons: [{ text: '' }]
}

export default EprocSetupFailure;
