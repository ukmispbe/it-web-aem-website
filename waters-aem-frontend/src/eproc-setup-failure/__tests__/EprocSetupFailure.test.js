import React from 'react';
import { shallow } from 'enzyme';

//Mocked Props
import EprocSetupFailure from '../EprocSetupFailure';
import Modal, { Header } from '../../utils/modal';
import AddToCartBody from '../../sku-details/views/addToCartModal';
import props from '../EprocSetupFailure.mock';

describe('<EprocSetupFailure />', () => {
    let enzymeWrapper;
    beforeAll(() => {
        enzymeWrapper = shallow(<EprocSetupFailure {...props} />);
    });
    it('should render without throwing an error', () => {
        expect(enzymeWrapper).toMatchSnapshot();
    });
    it('should display greeting, name, company, and logo', () => {
        expect(enzymeWrapper.find(Modal)).toHaveLength(1);
        expect(enzymeWrapper.find(Header)).toHaveLength(1);
        expect(enzymeWrapper.find(AddToCartBody)).toHaveLength(1);
        expect(enzymeWrapper.find(Modal).dive().html()).toBe('');
    });
    it('should display modal without close button', () => {
        enzymeWrapper.setProps({
            status: true,
            title: 'Sorry, something went wrong.',
            text: 'Please return to your procurement system and try again.',
            buttons: [{ text: 'RETURN TO PROCUREMENT SYSTEM', noAction: true, action: '#' }]
        });
        enzymeWrapper.update();
        const enzymeWrapperModal = enzymeWrapper.find(Modal).dive();
        expect(enzymeWrapperModal.find(Header)).toHaveLength(1);
        expect(enzymeWrapperModal.find(AddToCartBody)).toHaveLength(1);
    });
});