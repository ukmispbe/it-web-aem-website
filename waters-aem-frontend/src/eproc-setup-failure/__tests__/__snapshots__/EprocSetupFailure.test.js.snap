// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EprocSetupFailure /> should render without throwing an error 1`] = `ShallowWrapper {}`;
